src/SignalRServiceExtension/Config/ServiceManagerStore.cs

```diff
@@ -41,19 +43,25 @@ public IInternalServiceHubContextStore GetByConfigurationKey(string connectionSt
 
         private IInternalServiceHubContextStore CreateHubContextStore(string connectionStringKey)
         {
-            return new ServiceCollection().AddSignalRServiceManager()
-                .WithAssembly(Assembly.GetExecutingAssembly())
-                .SetupOptions<ServiceManagerOptions, OptionsSetup>(new OptionsSetup(configuration, loggerFactory, connectionStringKey))
-                .PostConfigure<ServiceManagerOptions>(o =>
+            var services = new ServiceCollection()
+                .PostConfigure<ContextOptions>(o =>
                 {
-                    if (string.IsNullOrWhiteSpace(o.ConnectionString))
+                    if (o.ServiceEndpoints == null || o.ServiceEndpoints.Length == 0)
                     {
                         throw new InvalidOperationException(ErrorMessages.EmptyConnectionStringErrorMessageFormat);
                     }
                 })
+                .AddSignalRServiceContext(new OptionsSetup(configuration, connectionStringKey))
+                .WithAssembly(Assembly.GetExecutingAssembly())
                 .AddSingleton(loggerFactory)
                 .AddSingleton(configuration)
-                .AddSingleton<IInternalServiceHubContextStore, ServiceHubContextStore>()
+                .AddSingleton<IServiceManager,ServiceManager>()  // for backward compatibility 
+                .AddSingleton<IInternalServiceHubContextStore, ServiceHubContextStore>();
+            if (router != null)
+            {
+                services.AddSingleton(router);
+            }
```

Comment:
Should use some default router if multiple endpoints are using?

Reply:
Default router is added inside SDK.